src/services/user/hooks/publicTeachers.js

```diff
@@ -18,8 +21,29 @@ const filterForPublicTeacher = (hook) => {
 	// Limit accessible user (only teacher which are discoverable)
 	hook.params.query.roles = ['teacher'];
 
-	if (process.env.IGNORE_DISCOVERABILITY !== 'true') {
-		hook.params.query.discoverable = true;
+	const TEACHER_VISIBILITY_FOR_EXTERNAL_TEAM_INVITATION =	hook.app.Config
+		.get('TEACHER_VISIBILITY_FOR_EXTERNAL_TEAM_INVITATION');
+	switch (TEACHER_VISIBILITY_FOR_EXTERNAL_TEAM_INVITATION) {
+		case 'opt-in':
+			// own school and other schools if opted-in
+			hook.params.query.$or = [
+				{ schoolId: hook.params.account.schoolId },
```

Comment:
imho it is questionable if you should ever see users of your own school in this dialog, but okay for now

Reply:
this is server, and generally should return public teachers, the client special case shoult take care on cleanup here